File: apps/web/src/lib/validators/vital.ts
/**
 * VitalSign Zod schemas.
 */
import { z } from "zod";

export const VitalSignSchema = z.object({
  patient_id: z.string().uuid(),
  encounter_id: z.string().uuid().optional(),
  heart_rate: z.number().min(0).max(300).optional(),
  systolic_bp: z.number().min(0).max(400).optional(),
  diastolic_bp: z.number().min(0).max(300).optional(),
  spo2: z.number().min(0).max(100).optional(),
  temperature: z.number().min(25).max(45).optional(),
  respiratory_rate: z.number().min(0).max(80).optional(),
  gcs: z.number().int().min(3).max(15).optional(),
  pain_score: z.number().int().min(0).max(10).optional(),
  source: z.enum(["manual", "device", "hl7"]).default("manual"),
});

export type VitalSignInput = z.infer<typeof VitalSignSchema>;

File: backend/src/domain/schemas/vital.py
"""VitalSign Pydantic schemas."""

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class VitalSignCreate(BaseModel):
    patient_id: uuid.UUID
    encounter_id: uuid.UUID | None = None
    recorded_at: datetime | None = None
    heart_rate: float | None = Field(None, ge=0, le=300)
    systolic_bp: float | None = Field(None, ge=0, le=400)
    diastolic_bp: float | None = Field(None, ge=0, le=300)
    spo2: float | None = Field(None, ge=0, le=100)
    temperature: float | None = Field(None, ge=25, le=45)
    respiratory_rate: float | None = Field(None, ge=0, le=80)
    gcs: int | None = Field(None, ge=3, le=15)
    pain_score: int | None = Field(None, ge=0, le=10)
    source: str = "manual"


class VitalSignResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    patient_id: uuid.UUID
    recorded_at: datetime
    source: str
    heart_rate: float | None
    systolic_bp: float | None
    diastolic_bp: float | None
    spo2: float | None
    temperature: float | None
    respiratory_rate: float | None
    gcs: int | None
    pain_score: int | None
